### DISCUSSION TITLE
Resample and smooth the vertices of triangle meshes

### DISCUSSION THREAD

**Discussion by zhengliuX** [pyvista/pyvista#2048] [Q&A]
Hi, is there any way that I could resample the triangle meshes?
Like if I have data with 30000 vertices, then I resample it to 15000, but it won't affect the shape of the mesh.
also, any way to smooth it to make the mesh much more smoothing?

I saw the tutorial about smooth [here](https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyDataFilters.smooth.html?highlight=smooth#pyvista.PolyDataFilters.smooth)
and might be resample [here](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.sample.html?highlight=sample#pyvista.DataSet.sample)
However, I'm not quite sure about how to do that on my own data.
Any suggestions?

BTW, the code about smooth https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyData.smooth.html?highlight=smooth#pyvista.PolyData.smooth seems not work for me, this is my result using this code.
![image](https://user-images.githubusercontent.com/65115609/150370861-7ab3c974-762b-4735-babd-b62a7e4650db.png)
Quite different from the doc.

**Comment by chautuankien** [comment]
i also have similar issue. Did you find the solution?

**Comment by darikg** [comment]
for reducing the number of vertices you could try [decimate](https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.PolyDataFilters.decimate.html#pyvista.PolyDataFilters.decimate) or [decimate_pro](https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.PolyDataFilters.decimate_pro.html)

not sure what's going on with smooth there, you're sure mesh had been subdivided as in the example code?

**Comment by banesullivan** [comment]
Perhaps a combination of ACVD and smoothing?:

- https://github.com/pyvista/pyacvd
- https://docs.pyvista.org/version/stable/examples/01-filter/surface-smoothing.html#sphx-glr-examples-01-filter-surface-smoothing-py
